src/utils.ts

```diff
@@ -116,3 +116,14 @@ export function invertMap<T1, T2>(map: Map<T1, T2>): Map<T2, T1> {
   }
   return returnValue;
 }
+
+/**
+ * Creates a Type Guard function for a string union type defined via an array as const.
+ */
+export function createStringUnionTypeGuardFunction<T extends string>(
```

Comment:
I don't understand what this doing.

Reply:
We have a lot of string literal types defined by arrays with the `as const` flag, especially in ascend.ts. This creates type guard functions for those types, using the array from which they are defined. 

In this particular PR, we never really use the type guard aspect of it, only the boolean return value. But I figured it's probably helpful to have a type guard function.